crop_health_api/__main__.py

```diff
@@ -106,6 +106,62 @@ async def docs():
     )
 
 
+@app.get("/ping")
+async def ping():
+    if settings.api_domain == "localhost":
+        torchserve_domain = "local_torchserve"
+    else:
+        torchserve_domain = "localhost"
+    try:
+        response = requests.get(f"http://{torchserve_domain}:8080/ping")
+        if response.status_code != 200:
+            raise HTTPException(status_code=response.status_code, detail=response.text)
+        return response.json()
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=str(e))
+
+
+@app.post("/predictions/single-HLT")
+async def singleHLT(request: Request):
+    return await torch_request(request, "single-HLT")
+
+
+@app.post("/predictions/multi-HLT")
+async def multiHLT(request: Request):
+    return await torch_request(request, "multi-HLT")
+
+
+@app.post("/predictions/binary")
+async def binary(request: Request):
+    return await torch_request(request, "binary")
+
+
+async def torch_request(request: Request, type):
+    if settings.api_domain == "localhost":
```

Comment:
This codeblock is duplicated in the ping-method. Can this be refactored to a common function?